nav2_behavior_tree/plugins/control/round_robin_node.cpp

```diff
@@ -74,7 +73,11 @@ BT::NodeStatus RoundRobinNode::tick()
     }
   }
 
-  halt();
+  // reset
```

Comment:
The `halt()` function in RoundRobbinNode is no longer called. Should it be removed, or should that be what's updated? Seems like they're pretty much the same except `   ControlNode::haltChildren(); ` is `ControlNode::halt();`. I think you should just change them in that function

Reply:
The design pattern I noticed in the behavior tree library was the for control nodes `halt()` is never called within the `tick()` function. `halt()` calls `haltChildren()` internally but it also sets the status of the current node back to `IDLE`. Within the `tick()` function, the control node is expected to halt its children on failure/success but not set itself back to `IDLE`. Calling the control node's `halt()` function is its parent's job and the `halt()` function is only expected to reset the node's internal variables.

Reply:
We override `halt` in this class and you removed the only use of it in the code, I think that's why the code coverage drops so much. The only difference you added and the function as written is 1 line that we could just update?